syntax = "proto3";

package masterworker;



service RpcImage {
    rpc SendImage (stream ImageRequest) returns (stream ImageResponse) {};
    rpc SendServerStream (stream FileChunk) returns (UploadStatus) {};
    rpc SendClientStream (ImageRequest) returns (stream ImageResponse) {};
}


message FileChunk {
    bytes content = 1;
}

enum UploadStatusCode {
    Unknown = 0;
    Ok = 1;
    Failed = 2;
}

message UploadStatus {
    string Message = 1;
    UploadStatusCode Code = 2;
}

message ImageResponse {
    string id = 1;
}

message ImageRequest {
    string filename = 1;
}





